.github/workflows/reusable.yml

```diff
@@ -0,0 +1,104 @@
+name: Deploy AKS-Construction
+
+on:
+  workflow_call:
+    inputs:
+      templateVersion:
+        description: 'Template Version'
+        required: false
+        default: '0.8.5-preview'
+        type: string
+      rg:
+        description: 'ResourceGroup name'
+        required: true
+        type: string
+      resourceName:
+        description: 'Unique name all your resources are based on'
+        required: true
+        type: string
+      templateParamFile:
+        description: 'Template Parameters file'
+        required: false
+        type: string
+      templateParams:
+        description: 'Template Paramterss, space seperated key=value'
+        required: true
```

Comment:
should be required=false if i'm not providing any params, which is the situation when i'm providing a parameters file

Reply:
ah, ok.  I guess we either need templateParams OR fileParams, so both need to be non-required :(